test/listener_test.coffee

```diff
@@ -117,21 +117,20 @@ describe 'Listener', ->
           testMessage = new TextMessage(@user, 'test')
           testMessage.match = sinon.stub().returns(true)
           testRegex = /test/
- 
+
           testListener = new TextListener(@robot, testRegex, callback)
           result = testListener.matcher(testMessage)
- 
+
           expect(result).to.be.ok
           expect(testMessage.match).to.have.been.calledWith(testRegex)
- 
+
         it 'does not match EnterMessages', ->
           callback = sinon.spy()
           testMessage = new EnterMessage(@user)
-          testMessage.match = sinon.stub().returns(true)
           testRegex = /test/
- 
+
           testListener = new TextListener(@robot, testRegex, callback)
           result = testListener.matcher(testMessage)
- 
+
           expect(result).to.not.be.ok
-          expect(testMessage.match).to.not.have.been.called
+          expect(callback).to.not.have.been.called
```

Comment:
Ah, now I understand. The intent of the test was to ensure that `match` wasn't accidentally getting called on non-`TextMessage` objects. If we are duck typing, I agree that the stub is wrong. If not, then the test can safely ensure that unknown methods aren't called.

Now that I understand the intent, I think the test can stay this way or change back. No strong feelings.